packages/language-server/src/manifest-handling.ts

```diff
@@ -54,6 +55,26 @@ export function getFlexEnabledFlagForXMLFile(xmlPath: string): boolean {
   return manifestData[closestManifestPath].flexEnabled;
 }
 
+export function getMinUI5VersionForXMLFile(
+  xmlPath: string
+): string | undefined {
+  const manifestFilesForCurrentFolder = filter(
+    Object.keys(manifestData),
+    (manifestPath) => xmlPath.startsWith(dirname(manifestPath))
+  );
+
+  const closestManifestPath = maxBy(
+    manifestFilesForCurrentFolder,
+    (manifestPath) => manifestPath.length
+  );
+
+  if (closestManifestPath === undefined) {
+    return undefined;
+  }
+
+  return manifestData[closestManifestPath].minUI5Version;
```

Comment:
`minUI5Version`comes from the manifest and this is a user controlled file.

Do we want / need special handling / logging if the user did not specify `minUI5Version`? or just implicitly return undefined?

Reply:
what happens if the user specified an invalid minUi5Version?
e.g:
- 666

or one that is not supported by the semantic model (very old one?).

Reply:
I tend for `undefined` because we default the version later and log the usage of the default version when the semantic model is being created. Otherwise we need to do the same for `flexEnabled` as well

Reply:
When an unsupported version is being used, then the model is empty and will be in memory cached. I can also verify this by reading the version information from CDN - I had this in mind but didn't do it so far...

Reply:
Now, the version is being negotiated using the version mapping information:
https://ui5.sap.com/version.json